.github/actions/dockerfiles/Dockerfile.alpine-binary

```diff
@@ -23,5 +23,18 @@ RUN case ${TARGETPLATFORM} in \
     && unzip ${BIN_ARCH}.zip -d /out
 
 FROM --platform=${TARGETPLATFORM} alpine
-COPY --from=builder /out/stacks-node /out/stacks-signer /bin/
-CMD ["stacks-node", "mainnet"]
+COPY --from=builder /out/* /bin/
+ARG TAG
+
+RUN case "${TAG}" in \
+        signer-*) \
+            echo "/bin/stacks-signer run --config /signer-config.toml" > /tmp/command.sh \
+            ;; \
+        *) \
+            echo "/bin/stacks-node mainnet" > /tmp/command.sh && \
+            rm /bin/blockstack-cli /bin/clarity-cli /bin/relay-server /bin/stacks-events /bin/stacks-inspect \
+            ;; \
+    esac && \
+    chmod +x /tmp/command.sh
+
+CMD ["sh", "-c", "/tmp/command.sh"]
```

Comment:
I feel if we're implementing logic in a Dockerfile to decide whether to run binary A or binary B, it's a good use case to just create a separate Dockerfile as it avoids introducing misdirection.

Reply:
fair point - the decision around this specifically was DRY. 
the singer/stacks-node dockerfiles are essentially identical, the only diff is the command that's being run (along with removing some unneeded binaries in the case of a signer image). 

room for improvement though, we can look into it later - could you file an issue on this and we'll explore it a bit more?

Reply:
[Created an issue for this here!](https://github.com/stacks-network/stacks-core/issues/4990)